### DISCUSSION TITLE
Testing Strategies

### DISCUSSION THREAD

**Discussion by brookslybrand** [remix-austin/remixaustin-com#8] [Ideas]
How do we want to test the site?

Related is #2, but there are some initial questions and infra we should setup.

@grantnorwood bootstrapped this project with the [Indie Stack](https://github.com/remix-run/indie-stack), which gives us [Cypress](https://www.cypress.io/) and [Jest](https://jestjs.io/) out the gate.

Personally, I'm all in on Jest + [Testing Library](https://testing-library.com/) for unit tests and component tests.

As far as Cypress, I think it's fine and the infra is there, so that's cool! However, I'm a little more excited about [Playwright](https://playwright.dev/) these days for end-to-end (e2e) testing.

Curious other's thoughts? Have an idea you'd rather show through code, open a PR!

**Comment by grantnorwood** [comment]
I've added some initial tests in both vitest & cypress _(https://github.com/remix-austin/remixaustin-com/pull/26, https://github.com/remix-austin/remixaustin-com/pull/30)_ because they were already instrumented in the Indie Stack, and it was a good opportunity for me to try something that's new to me _(vitest)_, as well as to learn more about the GitHub Actions to run everything via CI/CD.

However, it would be easy to implement another solution instead!  So I'm still totally open to discussing using Jest instead of vitest, and Playwright instead of Cypress.

I'd love for others to chime in!

**Reply by brookslybrand** [reply]
Awesome! Thanks so much for taking this on Grant.

I definitely see another talk coming up to share all your learnings on GitHub actions 🤓 

I'm good with vitest, I've been wanting to play with it more anyway and it seems super snappy. However, if jest is easier to implement, would love to hear it.

Cypress is fine, and I get that some of the boilerplate is already setup. Personally I'm much more interested in Playwright after seeing some coworkers use it. It targets more browsers which is already a big plus in my opinion.

**Reply by grantnorwood** [reply]
Well I'd like to get hands-on with Playwright, too.  And Cypress Cloud stole a few hours of my time troubleshooting the "record" functionality.  So let's give Playwright a shot!

I can do a talk on GitHub Actions, just let me know how long you'd like me to go.  Anything from lightning talk to full interactive workshop would be fun.

**Reply by grantnorwood** [reply]
I created PR https://github.com/remix-austin/remixaustin-com/pull/45 to add Playwright tests, but still working on getting the ci/cd to work properly.

Microsoft deprecated their [GitHub Action](https://github.com/microsoft/playwright-github-action) in favor of running CLI commands, but I gotta make sure the server starts and waits on the `200 OK` response before running the tests.

I'll try to get back to this in the next couple days!

**Reply by brookslybrand** [reply]
Awesome! Thanks Grant, appreciate you working on this CI/CD stuff. Hope you're having fun 😅

**Reply by grantnorwood** [reply]
Ok, I merged #45 to make sure ci/cd works when going to prod, and we're good to go!  I still want to publish the report as HTML/Markdown to the GH Workflow [status page](https://github.com/remix-austin/remixaustin-com/actions/runs/3645272672) instead of the downloadable .zip file artifact, but I'll get there.

<img width="400" alt="image" src="https://user-images.githubusercontent.com/707463/206358269-23f3fffb-b86e-4aaf-8259-cef7370daf8e.png">


If you've got add'l feedback on anything Playwright, just create a new issue or PR  🙂

 [![🚀 Deploy](https://github.com/remix-austin/remixaustin-com/actions/workflows/deploy.yml/badge.svg)](https://github.com/remix-austin/remixaustin-com/actions/workflows/deploy.yml)

**Comment by brookslybrand** [comment]
Calling bankruptcy on all discussions and issues today. Going to start revamping the direction of this project, so stay tuned 🚀
